internal/plugin/connectors/tcp/mssql/connector.go

```diff
@@ -33,69 +121,112 @@ func (connector *SingleUseConnector) Connect(
 	if err != nil {
-		connector.sendErrorToClient()
-		return nil, err
+		return connector.sendError("Unable to create new connection details: %s", err)
 	}
 
-	err = connector.ConnectToBackend(dataSourceName(connDetails))
+	// Create a new MsSQL connector
+	// Using DSN (Data Source Name) string because gomssql forces us to.
+	//
+	// NOTE: Secretless has some unfortunate naming collisions with the
+	// go-mssqldb driver package.  The driver package has its own concept of a
+	// "connector", and its connectors also have a "Connect" method.
+	driverConnector, err := mssql.NewConnector(dataSourceName(connDetails))
 	if err != nil {
-		connector.logger.Errorf("Failed to connect to backend: %s", err)
-		connector.sendErrorToClient()
-		return nil, err
+		return connector.sendError("Failed to create a go-mssqldb connector: %s", err)
 	}
 
-	// Now that the backend connection is authenticated, send the client
-	// a successful authentication response.
-	if _, err = clientConn.Write(connector.CreateAuthenticationOKMessage()); err != nil {
-		connector.logger.Errorf("Failed to send a successful authentication"+
-			" response to the client"+
-			": %s", err)
-		connector.sendErrorToClient()
-		return nil, err
-	}
+	// Create the context for our connection
+	ctx := context.Background()
 
-	return connector.backendConn, nil
-}
+	// Create a channel for receiving the prelogin response through the context
+	preLoginResponseChannel := make(chan map[uint8][]byte)
 
-// ConnectToBackend establishes the connection to the target database and sets
-// the backendConnection field.
-func (connector *SingleUseConnector) ConnectToBackend(dataSourceName string) error {
-	var err error
+	// Set a 'marker' for when the driver has finished connecting to the server
+	connectPhaseFinished := make(chan struct {})
 
-	// Using DSN (Data Source Name) string because gomssql forces us to.
-	//
-	// NOTE: Secretless has some unfortunate naming collisions with the
-	// go-mssqldb driver package.  The driver package has its own concept of a
-	// "connector", and its connectors also have a "Connect" method.
-	driverConnector, err := mssql.NewConnector(dataSourceName)
+	// Add channels to the context, to retrive information from the driver
+	loginContext := context.WithValue(ctx, keys.PreLoginResponseKey, preLoginResponseChannel)
+
+	// Build a new driver connection
+	var driverConn driver.Conn
+
+	go func() {
+		// Kick off authentication through our third party connector
+		driverConn, err = driverConnector.Connect(loginContext)
+		connectPhaseFinished <- struct{}{}
+
+		if err != nil {
+			_, err = connector.sendError("failed to connect to mssql server: %s", err)
+		}
+	} ()
+
+	// Blocks continuation until we've received the preLoginResponse from the driver
+	preloginResponse := <- preLoginResponseChannel
+
+	// This encryption setting should only apply to the client -> server, but since
+	// secretless is in-between, we want to ignore it and handle this ourselves
```

Comment:
This is correct now but we've sort of buried the lede.  Can you change to:
```
// Since the communication between the client and Secretless must be unencrypted,
// we fool the client into thinking that it's talking to a server that does not support
// encryption.  
```